### ISSUE TITLE
User defined Pose

### ISSUE THREAD

**Issue by ghost** [AliaksandrSiarohin/first-order-model#449]
Hello Author,

I am trying to implement hybrid of 1st order model and One shot neural talking head (Nvidia's one).

The idea is to have a user defined pose and the source should adapt to the given pose. One shot paper is doing the same but I want to modify 1st order pipeline to incorporate this.

Any tips and suggestion would be helpful :)

Best,
Vinayak

**Comment by AliaksandrSiarohin** [comment]
I don't understand what is user defined pose, how it is resented and how the user should get one?

**Comment by ghost** [comment]
Thanks for the quick reply. 

Here is the screenshot of one shot paper. Pose can be extracted by Head pose estimator network which gives yaw, pitch and roll. 
Ru, Tu (Rotation and Translation) is user defined whereas Rd, Td is from the estimator network. Page 13 of the below paper :)

Image Ref: https://arxiv.org/pdf/2011.15126.pdf
<img width="542" alt="Screenshot 2021-07-07 at 1 59 44 PM" src="https://user-images.githubusercontent.com/57089337/124756489-cadd7c00-df2c-11eb-90bc-dd23191819b6.png">

The user can modify the yaw, pitch and roll with respect to camera.

Looking forward to your reply :)

Best, Vinayak

**Comment by AliaksandrSiarohin** [comment]
Why not to use the same architecture than?

**Comment by ghost** [comment]
From your paper: _The purpose of the motion estimation module is to predict a dense motion field from a frame D of the driving video **D** to the source frame S. The dense motion field is later used to align the feature maps computed from S with the **object pose in D**_

I was thinking if Driving video could adapt to the pose of the Source frame. For an example if the the person looks up in the driving video, the pose in the source frame remains constant except for the expression part. Expression can be of the driving video.

**Comment by AliaksandrSiarohin** [comment]
Yes, so why can't you use their same architecture? In one of their ablation studies they do like this.

**Comment by ghost** [comment]
Thank you for your reply :) It simply takes too much of GPU power to train them. If I remember correctly they trained on cluster of GPUs with 32 GB memory. But Again, they can :)

**Comment by ghost** [comment]
Hey @AliaksandrSiarohin, I am following your suggestion and implementing one shot neural talking head :) 

Although, I have a question on FOMM. In the warp_coordinates method of class Transform, any reason for using Manhattan distance? 

It look forward to your reply :)

Best, Vinayak
